### PR TITLE
UCP/EP/FLUSH: fix error handling flow

### PR DESCRIPTION
**Pull request by evgeny-leksikov**

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
@yosefe pls take a look

File: src/ucp/rma/flush.c [hunk 1]
```diff
@@ -54,10 +55,20 @@ static void ucp_ep_flush_progress(ucp_request_t *req)
...
+             * - if count == 0, we have completed the flush on all lanes
+             * - otherwise, flush progress was re-scheduled from flush progress
+             *   pending right after ucp_worker_iface_err_handle_progress(),
+             *   so remove destroyed/failed lanes from started_lanes and count
+             *   them completed.
              */
+            ucs_assert(ep->flags & UCP_EP_FLAG_FAILED);
+            if (req->send.state.uct_comp.count > 0) {
+                destroyed_lanes = req->send.flush.started_lanes & ~all_lanes;
+                req->send.flush.started_lanes  &= ~destroyed_lanes;
```
Comment (yosefe) [inline src/ucp/rma/flush.c:72]: maybe &= all_lanes ?

File: src/ucp/rma/flush.c [hunk 2]
```diff
@@ -54,10 +55,20 @@ static void ucp_ep_flush_progress(ucp_request_t *req)
...
+                req->send.state.uct_comp.count -= ucs_popcount(destroyed_lanes);
```
Comment (yosefe) [inline src/ucp/rma/flush.c:73]: align on =
add ucs_debug that "req %p: lanes 0x%x were destroyed so reducing comp count by %d"

File: src/ucp/rma/flush.c (hunk 1 above)
Comment (evgeny-leksikov) [reply src/ucp/rma/flush.c:72]: anyway need to calculate `destroyed_lanes` for counter update, so IMO `~destroyed_lanes` looks clearer - aligned with comment above and counter update

File: src/ucp/rma/flush.c (hunk 2 above)
Comment (evgeny-leksikov) [reply src/ucp/rma/flush.c:73]: > align on =

lines becomes too long, separated with `ucs_debug`

**Review by yosefe — APPROVED** [review]

**Pull request merged**